### PR TITLE
Cli restructure

### PR DESCRIPTION
**Pull request by VirajSalaka**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java [hunk 1]
```diff
@@ -59,18 +75,70 @@ public void execute() {
...
 
+        projectName = GatewayCmdUtils.getProjectName(mainArgs);
+        projectName = projectName.replaceAll("[\\/\\\\]", "");
+        File projectLocation = new File(GatewayCmdUtils.getProjectDirectoryPath(projectName));
+
+        if (!projectLocation.exists()) {
+            throw new CLIRuntimeException("Project " + projectName + " does not exist.");
+        }
+
+        if(!isCompiled){
```
Comment (RAVEENSR) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java:86]: Add a space after closing parenthesis

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java [hunk 2]
```diff
@@ -59,18 +75,70 @@ public void execute() {
...
+            try{
```
Comment (RAVEENSR) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java:87]: Add space before {. Fix spaces related errors in other places as well(ex: Line 86,115,126,135).

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java [hunk 3]
```diff
@@ -26,17 +26,10 @@
...
 import org.wso2.apimgt.gateway.cli.constants.GatewayCliConstants;
 import org.wso2.apimgt.gateway.cli.constants.RESTServiceConstants;
-import org.wso2.apimgt.gateway.cli.exception.BallerinaServiceGenException;
-import org.wso2.apimgt.gateway.cli.exception.CLIInternalException;
-import org.wso2.apimgt.gateway.cli.exception.CLIRuntimeException;
-import org.wso2.apimgt.gateway.cli.exception.CliLauncherException;
-import org.wso2.apimgt.gateway.cli.exception.ConfigParserException;
-import org.wso2.apimgt.gateway.cli.exception.HashingException;
-import org.wso2.apimgt.gateway.cli.hashing.HashUtils;
+import org.wso2.apimgt.gateway.cli.exception.*;
```
Comment (RAVEENSR) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java:32]: Remove wild card imports. Fix in other places as well(Line 49)

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java [hunk 4]
```diff
@@ -219,32 +214,47 @@ public void execute() {
...
+                            "         \"prod\": {\n" +
+                            "            \"type\": \"load_balance\",\n" +
+                            "            \"endpoints\": [\n" +
+                            endpoint.trim() +
+                            "            ]\n" +
+                            "         }\n" +
+                            "      }";
+                }
+
+                if(StringUtils.isEmpty(basepath)){
```
Comment (RAVEENSR) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java:245]: Formatting issue. (In line 247 as well)

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/codegen/CodeGenerator.java [hunk 5]
```diff
@@ -21,18 +21,18 @@
...
+import org.wso2.apimgt.gateway.cli.exception.CLIInternalException;
+import org.wso2.apimgt.gateway.cli.exception.CLIRuntimeException;
 import org.wso2.apimgt.gateway.cli.model.rest.ext.ExtendedAPI;
 import org.wso2.apimgt.gateway.cli.model.template.GenSrcFile;
 import org.wso2.apimgt.gateway.cli.model.template.service.BallerinaService;
 import org.wso2.apimgt.gateway.cli.model.template.service.ListenerEndpoint;
-import org.wso2.apimgt.gateway.cli.utils.CodegenUtils;
-import org.wso2.apimgt.gateway.cli.utils.GatewayCmdUtils;
-import org.wso2.apimgt.gateway.cli.utils.OpenApiCodegenUtils;
+import org.wso2.apimgt.gateway.cli.utils.*;
```
Comment (RAVEENSR) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/codegen/CodeGenerator.java:35]: Avoid wild card imports

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java [hunk 6]
```diff
@@ -219,32 +214,47 @@ public void execute() {
...
+                    basepath = SwaggerUtils.getBasePathFromSwagger(apiDefPath);
+                    if(StringUtils.isEmpty(basepath)){
+                        if ((basepath = promptForTextInput( "Enter basepath: "))
+                                .trim().isEmpty()) {
```
Comment (RAVEENSR) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java:249]: Move this section to previous line

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/codegen/ThrottlePolicyGenerator.java [hunk 7]
```diff
@@ -42,24 +47,66 @@
...
 
-        List<GenSrcFile> genSubsFiles = generateSubscriptionPolicies(subscriptionPolicies);
-        genFiles.addAll(genSubsFiles);
+    //todo: remove unnecessary code segments
+    public void generate(String outPath, String projectName) throws IOException {
+
+        ApplicationThrottlePolicyListDTO applicationPolicies = restoreApplicationThrottlePolicy(projectName);
+        SubscriptionThrottlePolicyListDTO subscriptionPolicies = restoreSubscriptionThrottlePolicy(projectName);
+
+        if(applicationPolicies == null && subscriptionPolicies == null){
```
Comment (RAVEENSR) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/codegen/ThrottlePolicyGenerator.java:86]: Formatting issue. Add relevant spaces. Fix this issue in other files as well.

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/rest/RESTAPIServiceImpl.java [hunk 8]
```diff
@@ -27,17 +27,14 @@
 import org.wso2.apimgt.gateway.cli.exception.CLIRuntimeException;
 import org.wso2.apimgt.gateway.cli.model.config.Config;
 import org.wso2.apimgt.gateway.cli.model.config.MutualSSL;
-import org.wso2.apimgt.gateway.cli.model.rest.APIListDTO;
-import org.wso2.apimgt.gateway.cli.model.rest.ClientCertMetadataDTO;
-import org.wso2.apimgt.gateway.cli.model.rest.ClientCertificatesDTO;
-import org.wso2.apimgt.gateway.cli.model.rest.Endpoint;
-import org.wso2.apimgt.gateway.cli.model.rest.EndpointConfig;
+import org.wso2.apimgt.gateway.cli.model.rest.*;
```
Comment (RAVEENSR) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/rest/RESTAPIServiceImpl.java:30]: Avoid wild card import

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java (hunk 1 above)
Comment (VirajSalaka) [reply components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java:86]: code segment is no longer available.

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java (hunk 2 above)
Comment (VirajSalaka) [reply components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java:87]: code segment is no longer available.

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java (hunk 3 above)
Comment (VirajSalaka) [reply components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java:32]: code segment is no longer available.

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java (hunk 4 above)
Comment (VirajSalaka) [reply components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java:245]: code segment is no longer available.

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/codegen/CodeGenerator.java (hunk 5 above)
Comment (VirajSalaka) [reply components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/codegen/CodeGenerator.java:35]: Already fixed by a later added PR.

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java (hunk 6 above)
Comment (VirajSalaka) [reply components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java:249]: code segment is no longer available.

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/codegen/ThrottlePolicyGenerator.java (hunk 7 above)
Comment (VirajSalaka) [reply components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/codegen/ThrottlePolicyGenerator.java:86]: Already fixed by a later added PR.

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/rest/RESTAPIServiceImpl.java (hunk 8 above)
Comment (VirajSalaka) [reply components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/rest/RESTAPIServiceImpl.java:30]: Already fixed by a later added PR.

**Pull request merged**